src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -4735,6 +4735,86 @@ CORINFO_FIELD_HANDLE MethodContext::repGetFieldInClass(CORINFO_CLASS_HANDLE clsH
     return result;
 }
 
+void MethodContext::recGetTypeLayout(GetTypeLayoutResult result, CORINFO_CLASS_HANDLE typeHnd, CORINFO_TYPE_LAYOUT_NODE* nodes, size_t numNodes)
+{
+    if (GetTypeLayout == nullptr)
+        GetTypeLayout = new LightWeightMap<DWORDLONG, Agnostic_GetTypeLayoutResult>();
+
+    DWORDLONG key = CastHandle(typeHnd);
+
+    int index = GetTypeLayout->GetIndex(key);
+    if (index != -1)
+    {
+        // If we already have this then just skip.
+        Agnostic_GetTypeLayoutResult existing = GetTypeLayout->GetItem(index);
+        if ((existing.result != (DWORD)GetTypeLayoutResult::Partial) ||
+            (result == GetTypeLayoutResult::Failure) ||
+            (numNodes <= existing.numNodes))
+        {
+            // No new data to add
+            return;
+        }
+
+        // Otherwise fall through and update the map with more information.
+    }
+
+    Agnostic_GetTypeLayoutResult value;
+    ZeroMemory(&value, sizeof(value));
+
+    value.result = (DWORD)result;
+    if (result == GetTypeLayoutResult::Failure)
+    {
+        value.nodesBuffer = UINT_MAX;
+    }
+    else
+    {
+        Agnostic_CORINFO_TYPE_LAYOUT_NODE* agnosticFields = new Agnostic_CORINFO_TYPE_LAYOUT_NODE[numNodes];
+        for (size_t i = 0; i < numNodes; i++)
+        {
+            agnosticFields[i] = SpmiRecordsHelper::StoreAgnostic_CORINFO_TYPE_LAYOUT_NODE(nodes[i]);
+        }
+
+        value.nodesBuffer = GetTypeLayout->AddBuffer((unsigned char*)agnosticFields, (unsigned int)(sizeof(Agnostic_CORINFO_TYPE_LAYOUT_NODE) * numNodes));
+        value.numNodes = (DWORD)numNodes;
+
+        delete[] agnosticFields;
+    }
+
+    if (index != -1)
+    {
+        GetTypeLayout->Update(index, value);
+    }
+    else
+    {
+        GetTypeLayout->Add(key, value);
+    }
+}
+void MethodContext::dmpGetTypeLayout(DWORDLONG key, const Agnostic_GetTypeLayoutResult& value)
+{
+    printf("FlattenType key type-%016" PRIX64 " value result=%d numNodes=%d", key, (DWORD)value.result, value.numNodes);
```

Comment:
This function should dump all the stored `CORINFO_TYPE_LAYOUT_NODE` (with their indices) also.